src/main/java/com/limechain/runtime/hostapi/StorageHostFunctions.java

```diff
@@ -98,10 +100,10 @@ public List<ImportObject> buildFunctions() {
      * @param valuePointer a pointer-size containing the key.
      */
     public void extStorageSetVersion1(RuntimePointerSize keyPointer, RuntimePointerSize valuePointer) {
-        byte[] key = hostApi.getDataFromMemory(keyPointer);
+        Nibbles key = Nibbles.fromBytes(hostApi.getDataFromMemory(keyPointer));
```

Comment:
Have you verified this is the expected behaviour? I know the spec is lacking in details here, so I'd suppose you've cross-referenced with other implementations?